eng/Versions.props

```diff
@@ -26,21 +26,21 @@
   </PropertyGroup>
   <!-- Production Dependencies -->
   <PropertyGroup>
-    <SystemCollectionsImmutableVersion>6.0.0</SystemCollectionsImmutableVersion>
-    <SystemConfigurationConfigurationManagerVersion>6.0.0</SystemConfigurationConfigurationManagerVersion>
+    <SystemCollectionsImmutableVersion>7.0.0</SystemCollectionsImmutableVersion>
+    <SystemConfigurationConfigurationManagerVersion>7.0.0</SystemConfigurationConfigurationManagerVersion>
     <!--
         Modifying the version of System.Memory is very high impact and causes downstream breaks in third-party tooling that uses the MSBuild API.
         When updating the version of System.Memory file a breaking change here: https://github.com/dotnet/docs/issues/new?assignees=gewarren&labels=breaking-change%2CPri1%2Cdoc-idea&template=breaking-change.yml&title=%5BBreaking+change%5D%3A+
         and follow the guidelines written here (internal-link): https://dev.azure.com/devdiv/DevDiv/_wiki/wikis/DevDiv.wiki/1796/How-to-add-a-Known-Issue
     -->
     <SystemMemoryVersion>4.5.5</SystemMemoryVersion>
     <SystemNetHttpVersion>4.3.4</SystemNetHttpVersion>
-    <SystemReflectionMetadataLoadContextVersion>6.0.0</SystemReflectionMetadataLoadContextVersion>
-    <SystemReflectionMetadataVersion>6.0.0</SystemReflectionMetadataVersion>
-    <SystemResourcesExtensionsPackageVersion>6.0.0</SystemResourcesExtensionsPackageVersion>
-    <SystemSecurityPermissionsVersion>6.0.0</SystemSecurityPermissionsVersion>
+    <SystemReflectionMetadataLoadContextVersion>7.0.0</SystemReflectionMetadataLoadContextVersion>
+    <SystemReflectionMetadataVersion>7.0.0</SystemReflectionMetadataVersion>
```

Comment:
@ViktorHofer, @jkoritzinsky, @trylek re: https://github.com/dotnet/arcade/issues/10516

Should we also update this in host model? https://github.com/dotnet/runtime/blob/1f8d401995fe904c69d28a05bef7948d6affe2c2/src/installer/managed/Microsoft.NET.HostModel/Microsoft.NET.HostModel.csproj#L22-L23 @mmitche has updated it in the SDK repo.

This came up yesterday in another discussion https://github.com/dotnet/runtime/pull/85038#discussion_r1174593791. I think we can unify SRM version to 7.0 across the runtime repo (provided Tomas' ILSpy concern is addressed).

Reply:
Could we re-target HostModel to net7.0/net8.0? That would solve the PackageReference issue and would remove the prebuild on S.T.J and S.R.M.

Reply:
Yeah Microsoft.NET.HostModel is used by the sdk to implement the tasks that manipulate the apphosts. The dependency requirements for that package are that the dependencies are compatible with being loaded by an MSBuild task in dotnet/sdk build/publish scenarios. We can make any changes that are compatible with that rule (including package upgrades/downgrades, etc).